code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -144,19 +144,27 @@
 
 /mob/living/silicon/robot/drone/construction/matriarch/Initialize()
 	. = ..()
-	SSghostroles.add_spawn_atom("matriarchmaintdrone", src)
+	if(SSticker.current_state == GAME_STATE_PLAYING)
+		request_player()
+	else
+		LAZYADD(SSatoms.late_misc_firers, src)
 
 /mob/living/silicon/robot/drone/construction/matriarch/assign_player(mob/user)
 	. = ..()
 	SSghostroles.remove_spawn_atom("matriarchmaintdrone", src)
 
 /mob/living/silicon/robot/drone/construction/matriarch/ghostize(can_reenter_corpse, should_set_timer)
 	. = ..()
-	SSghostroles.add_spawn_atom("matriarchmaintdrone", src)
+	if(src in mob_list) // needs to exist to reopen spawn atom
```

Comment:
for anyone curious, calling Destroy removes it from the mob_list, which is why this check works